### PR TITLE
Model datasetup

### PR DESCRIPTION
**Pull request by lordcod99**
@surendhar-palanisamy can you please review this and can u also mention abut the root path I need to use here : https://github.com/ai4ce/UNavWebsocket/blob/b767669859ba75d68594e6d15593024dab851b37/src/modal_server.py#L29

### CODE REVIEW

**Comment by surendhar-palanisamy** [comment]
@lordcod99 

example code to mount local repo's
image

File: src/modal_server.py [hunk 1]
```diff
@@ -26,6 +26,8 @@
 # Inactivity timeout in minutes
 INACTIVITY_TIMEOUT = 30
 
+data_handler = DataHandler('root') # HERE PLEASE MODIFY THR root TO RESPECTIVE VALUE
```
Comment (pramttl) [inline src/modal_server.py:29]: Use modal volumes for this: https://modal.com/docs/guide/volumes, for the modal setup.

File: src/modal_server.py [hunk 2]
```diff
@@ -174,6 +176,8 @@ def monitor_inactivity():
 @app.function(image=custom_image)
 @modal.wsgi_app()
 def run_server():
+    data_handler.download_data()
```
Comment (pramttl) [inline src/modal_server.py:179]: We don't want to do this download on every server start, this should be one time (during image build) and cached in volumes.

Also, every image build after a dependency changes shouldn't refetch the data, it should use what's cached in the volume.

File: src/utils/data_handler.py [hunk 3]
```diff
@@ -15,6 +22,60 @@ def __init__(self, new_root_dir):
...
+                logging.info(f"Downloaded {destination}")
+            else:
+                logging.info(f"{destination} already exists. Skipping download.")
+
+        for filename, file_id in files.items():
+            logging.info(f"Processing {os.path.join(self.new_root_dir, filename)}")
+            file_path = os.path.join(self.new_root_dir, filename)
+            download_file_from_google_drive(file_id, file_path)
+
+    def rearrange_data(self):
```
Comment (pramttl) [inline src/utils/data_handler.py:53]: Let's give a more explicit name or add a docstring on why we need to rearrange.

Comment (pramttl) [reply src/utils/data_handler.py:53]: nit: Also, let's prefix intended private methods with `_`.

**Comment by surendhar-palanisamy** [comment]
@lordcod99 

here is example of image building process

image  = (
    modal.Image.debian_slim(python_version="3.8")
    .run_commands(
        "apt-get update",
        "apt-get install -y pkg-config libhdf5-dev gcc libgl1-mesa-glx libglib2.0-0",
        "pip install --upgrade pip",
    )
        .run_commands(
         "apt-get update",
        "apt-get install -y build-essential python3 python3-pip git cmake libeigen3-dev libgoogle-glog-dev libgflags-dev libatlas-base-dev libeigen3-dev libsuitesparse-dev pkg-config libhdf5-dev gcc libgl1-mesa-glx libglib2.0-0",
        "apt-get remove -y libeigen3-dev",  # Remove the existing version of Eigen3
    )
    .run_commands(
        "apt-get install -y  cmake libsuitesparse-dev libgoogle-glog-dev libgflags-dev libatlas-base-dev libeigen3-dev"
    )
   .run_commands(
       "apt-get install -y libceres-dev"
   )
   .run_commands(
       "git clone https://gitlab.com/libeigen/eigen.git eigen"
   )
   .workdir("/eigen")
   .run_commands(
       "git checkout 3.4",
        "mkdir build",
   )
   .workdir("/eigen/build")
   .run_commands(
        "cmake ..",
        "make",
        "make install",
   )
   .workdir("/")
   .run_commands("ls")
    .run_commands(
        "git clone https://github.com/cvg/implicit_dist.git implicit_dist",
        )
    .workdir("/implicit_dist")
    .run_commands(
        "ls",
        "python3 -m venv .venv",
        ". .venv/bin/activate",
        "pip install .",
        "pip freeze",
    )
    .workdir('/root')
    .run_commands("ls")
    .pip_install_from_requirements("src/requirements.txt")
    .pip_install("kornia")
    .pip_install('pytorch_lightning')

**Review by pramttl — CHANGES_REQUESTED** [review]

**Comment by lordcod99** [comment]
@surendhar-palanisamy @pramttl update the pr with new commit please review

File: src/modal_server.py (hunk 1 above)
Comment (lordcod99) [reply src/modal_server.py]: used modal volumes in new update

File: src/modal_server.py (hunk 2 above)
Comment (lordcod99) [reply src/modal_server.py]: changed it to one time set up script

File: src/modalServer_dataSetup.py [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+import os
+import shutil 
+import logging
+import yaml
+import zipfile
+
+logging.basicConfig(level=logging.INFO, format='%(asctime)s - %(levelname)s - %(message)s')
+
+image = modal.Image.debian_slim().run_commands(
+    "pip install gdown",
```
Comment (pramttl) [inline src/modalServer_dataSetup.py:12]: We could move these to the requirements.txt file?

**Pull request closed without merging**